components/mediation/tasks/org.wso2.micro.integrator.ntask.core/src/main/java/org/wso2/micro/integrator/ntask/core/impl/AbstractQuartzTaskManager.java

```diff
@@ -282,7 +282,7 @@ protected synchronized void resumeLocalTask(String taskName) throws TaskExceptio
         }
     }
 
-    private boolean containsLocalTask(String taskName, String taskGroup) throws TaskException {
+    protected boolean isPreviouslyScheduled(String taskName, String taskGroup) throws TaskException {
```

Comment:
what does previosulyScheduled mean?

Reply:
The task was already scheduled and ran in this node and has been stopped for some reason.

Reply:
previouslyScheduledButFailed then? I didn't understand it previously, that's why

Reply:
its not failed , but has been stopped . The method name isPreviouslyScheduled is precise, it gives the result whether the task has been scheduled already in this node already or not.